ops/charm.py

```diff
@@ -190,6 +217,8 @@ def __init__(self, raw=None):
         self.payloads = {name: PayloadMeta(name, payload)
                          for name, payload in raw.get('payloads', {}).items()}
         self.extra_bindings = raw.get('extra-bindings', [])
+        self.functions_type = raw.get('functions_type', None)
```

Comment:
This was added as a replacement for the logic I had before that passed `event_dir` to `_create_event_link`:
https://github.com/canonical/operator/pull/79/commits/b4245651e1f4db2c47fd4d6f4681f29f22f8b3a0#diff-2a757bffeaf9ecf9b797cba4ea3223bbL83-L90

It would be good to discuss if I should get back to the old logic or come up with some other alternative.